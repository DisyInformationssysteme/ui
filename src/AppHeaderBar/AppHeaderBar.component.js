import React from 'react';
import { Link } from 'react-router';
import Dropdown from 'react-bootstrap/lib/Dropdown';
import MenuItem from 'react-bootstrap/lib/MenuItem';
import { api, Dispatcher, Icon } from 'react-cmf';
import { themr } from 'react-css-themr';
import { APP_HEADER_BAR } from '../identifiers';
// TODO: we have an issue during the build on this.


function getActions(context, id) {
	if (typeof id === 'string') {
		return api.action.getActionInfo(context, id);
	}
	return id;
}

/**
 * This is the main bar for webapp
 * @param {object} props   react props
 * @param {object} context react context
 */
function AppHeaderBar(props, context) {

	const actions = [];
	if (Array.isArray(props.actions)) {
		props.actions.forEach((action) => {
			actions.push(getActions(context, action));
		});
	}
	const userMenuActions = [];
	if (Array.isArray(props.userMenuActions)) {
		props.userMenuActions.forEach((action) => {
			userMenuActions.push(getActions(context, action));
		});
	}
	return (
		<div className={props.theme.appHeaderBar}>
			<div className={props.theme.brand}>
				<Link to="/" className={props.theme.link}>
					{props.logo ? <img
						className={props.theme.logo}
						src={props.logo.src}
						alt={props.logo.alt}
						style={props.logo.style}
					/> : null}
					{props.app}
				</Link>
			</div>
			<div className={props.theme.links}>
				{actions.map((action, i) => (
					<Dispatcher onClick={action} key={i}>
						<a className={props.theme.link}>
							<Icon name={action.icon} className="fa-fw" />
						</a>
					</Dispatcher>
				))}
				<Dropdown id="usermenu" pullRight>
					<Dropdown.Toggle useAnchor className="btn btn-inverse">
						<Icon name="icon-profile" className="fa-fw" />
						Micke TUCHEN
					</Dropdown.Toggle>
					<Dropdown.Menu >
						{userMenuActions.map((action, i) => (
							<Dispatcher onClick={action} key={i}>
								<MenuItem eventKey="1">
									<Icon name={action.icon} className="fa-fw" />
									{action.name}
								</MenuItem>
							</Dispatcher>
						))}
					</Dropdown.Menu>
				</Dropdown>
			</div>
		</div>
	);
}

AppHeaderBar.propTypes = {
	actions: React.PropTypes.arrayOf(React.PropTypes.string),
	userMenuActions: React.PropTypes.arrayOf(React.PropTypes.string),
	logo: React.PropTypes.shape({
		src: React.PropTypes.string,
		alt: React.PropTypes.string,
		style: React.PropTypes.string,
	}),
	app: React.PropTypes.string,
	theme: React.PropTypes.shape({
		appHeaderBar: React.PropTypes.string,
		brand: React.PropTypes.string,
		link: React.PropTypes.string,
		links: React.PropTypes.string,
		logo: React.PropTypes.string,
	}),
};
AppHeaderBar.contextTypes = {
	store: React.PropTypes.object,
};
export default themr(APP_HEADER_BAR, null)(AppHeaderBar);
export { AppHeaderBar };
