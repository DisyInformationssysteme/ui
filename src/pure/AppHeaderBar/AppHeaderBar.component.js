import React from 'react';
import { themr } from 'react-css-themr';
import classNames from 'classnames';
import { APP_HEADER_BAR } from '../../identifiers';

/**
 * The top bar is the place where the user finds useful information and tools
 * to describe the application
 * the logo, the name of the application, an optionnal global search,
 * few icons for feedback & onboarding and a menu to access logout
 * and profile information.
 * @param {object} props   react props
 */
function AppHeaderBar(props) {
	let logo = null;
	if (props.logo) {
		logo = (<img
			className={props.theme.logo}
			src={props.logo.src}
			alt={props.logo.alt}
			style={props.logo.style}
		/>);
	}
	let link = (
		<a href="#/" className={props.theme.link}>
			{logo} {props.app}
		</a>
	);
	if (props.link) {
		link = (
			<props.link>{logo} {props.app}</props.link>
		);
	}
	const wrapper = classNames(props.theme.appHeaderBar, 'navbar');
	const brand = classNames(props.theme.brand, 'navbar-brand');
	return (
		<div className={wrapper}>
			<div className={brand}>
				{link}
			</div>
			{props.children}
		</div>
	);
}

AppHeaderBar.propTypes = {
	logo: React.PropTypes.shape({
		src: React.PropTypes.string,
		alt: React.PropTypes.string,
		style: React.PropTypes.object,
	}),
	app: React.PropTypes.string,
	theme: React.PropTypes.shape({
		appHeaderBar: React.PropTypes.string,
		brand: React.PropTypes.string,
		link: React.PropTypes.string,
		logo: React.PropTypes.string,
	}),
};
export default themr(APP_HEADER_BAR)(AppHeaderBar);
export { AppHeaderBar as PureAppHeaderBar };
