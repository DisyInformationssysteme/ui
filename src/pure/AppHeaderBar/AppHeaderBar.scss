$tlnd-app-menu-bg-color: #465A65 !default;
$tlnd-app-menu-txt-color: white !default;

// AppHeaderBar styles
.appHeaderBar {
	background-color: $tlnd-app-menu-bg-color;
	height: 100%;
}
.brand {
	display: inline-block;
	width: 50%
}
.link, .link:hover {
	color: $tlnd-app-menu-txt-color;
	cursor: pointer;
	text-decoration: none;
	margin: 10px;
}

.links {
	display: inline-block;
	width: 50%;
	text-align: right;
	padding: 12px 20px 0;
}
.logo {
  max-width: '100px';
}
