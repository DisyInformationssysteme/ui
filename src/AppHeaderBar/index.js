import { themr } from 'react-css-themr';
import { APP_HEADER_BAR } from '../identifiers';
import AppHeaderBar from './AppHeaderBar.component';
import theme from './AppHeaderBar.scss';

const ThemedAppHeaderBar = themr(APP_HEADER_BAR, theme)(AppHeaderBar);

export default ThemedAppHeaderBar;
export { ThemedAppHeaderBar as AppHeaderBar };
