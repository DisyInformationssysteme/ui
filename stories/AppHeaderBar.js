import React from 'react';
import { storiesOf, action } from '@kadira/storybook';

import { AppHeaderBar } from '../src';

const props = {
	app: 'Example App Name',
	brandLink: {
		onClick: action('brandonClick'),
	},
	content: [
		{
			navs: [
				{
					navItems: [
						{
							type: 'navItem',
							item: { icon: 'fa fa-bars', name: 'hello', onClick: action('onClick bars') },
						},
					],
				},
				{
					nav: { pullRight: true },
					navItems: [
						{
							type: 'navItem',
							item: { icon: 'fa fa-heart', name: 'world', onClick: action('onClick heart') },
						},
						{
							type: 'dropdown',
							item: {
								dropdown: {
									id: 'myDropdown',
									title: 'user 1',
									onSelect: action('dropdown onSelect'),
								},
								items: [
									{ icon: 'fa fa-fw fa-cog', name: 'settings', onClick: action('onClick settings') },
								],
							},
						},
					],
				},
			],
		},
		{
			forms: [{
				form: {
					pullRight: true,
					onSubmit: action('onSubmit'),
					role: 'search',
				},
				formgroups: [
					{
						formgroup: {
							controlId: 'globalSearch',
						},
						formcontrol: {
							type: 'text',
							placeholder: 'search anything',
							onChange: action('onChange in search'),
						},
					},
				],
				button: {
					onClick: action('onClick form'),
					bsStyle: 'link',
				},
				icon: 'fa fa-search',
			}],
		},
	],
};

storiesOf('App Header Bar', module)
	.addWithInfo('default', () => (
		<div>
			<AppHeaderBar {...props} />
			<div className="container" style={{ paddingTop: 40 }}>
				<h1>AppHeaderBar</h1>
				<h2>Definition</h2>
				<p>Display a navigation bar on top of the page.</p>
				<h2>Examples</h2>
				<p>Look on top</p>
				<pre>{JSON.stringify(props, null, 2)}</pre>
			</div>
		</div>
	))
	.addWithInfo('without brandLink', () => {
		const myprops = Object.assign({}, props);
		delete myprops.brandLink;
		return (
			<div>
				<AppHeaderBar {...myprops} />
				<div className="container" style={{ paddingTop: 40 }}>
					<h1>AppHeaderBar</h1>
					<h2>Definition</h2>
					<p>Display a navigation bar on top of the page.</p>
					<p>Look on top</p>
				</div>
			</div>
		);
	});
