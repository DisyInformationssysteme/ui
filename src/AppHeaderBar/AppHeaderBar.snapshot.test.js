import React from 'react';
import renderer from 'react-test-renderer';
import AppHeaderBar from './AppHeaderBar.component';

jest.mock('react-dom');

const props = {
	app: 'Example App Name',
	brandLink: {
		title: 'my brand title',
		className: 'my-brand-class',
		onClick: jest.fn(),
	},
	content: [
		{
			navs: [
				{
					navItems: [
						{
							type: 'navItem',
							item: { icon: 'fa fa-bars', name: 'hello', onClick: jest.fn() },
						},
					],
				},
				{
					nav: { pullRight: true },
					navItems: [
						{
							type: 'navItem',
							item: { icon: 'fa fa-heart', name: 'world', onClick: jest.fn() },
						},
						{
							type: 'dropdown',
							item: {
								dropdown: {
									id: 'myDropdown',
									title: 'user 1',
									onSelect: jest.fn(),
								},
								items: [
									{ icon: 'fa fa-fw fa-cog', name: 'settings', onClick: jest.fn() },
								],
							},
						},
					],
				},
			],
		},
		{
			forms: [{
				form: {
					pullRight: true,
					onSubmit: jest.fn(),
				},
				formgroups: [
					{
						formgroup: {
							controlId: 'globalSearch',
						},
						formcontrol: {
							type: 'text',
							placeholder: 'search anything',
							onChange: jest.fn(),
						},
					},
				],
				button: {
					onClick: jest.fn(),
				},
				buttonLabel: 'Save',
				icon: 'fa fa-search',
			}],
		},
	],
};

describe('AppHeaderBar', () => {
	it('should render', () => {
		// given
		const appBar = <AppHeaderBar {...props} />;

		// when
		const appBarInstance = renderer.create(appBar).toJSON();

		// then
		expect(appBarInstance).toMatchSnapshot();
	});
	it('should render wihtout brandLink', () => {
		// given
		const myprops = Object.assign({}, props);
		delete myprops.brandLink;
		const appBar = <AppHeaderBar {...myprops} />;

		// when
		const appBarInstance = renderer.create(appBar).toJSON();

		// then
		expect(appBarInstance).toMatchSnapshot();
	});
});
