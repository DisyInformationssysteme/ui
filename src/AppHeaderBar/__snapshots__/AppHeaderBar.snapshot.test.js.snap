exports[`AppHeaderBar should render 1`] = `
<nav
  className="tc-app-header-bar navbar navbar-inverse navbar-fixed-top">
  <div
    className="container-fluid">
    <div
      className="navbar-header">
      <button
        className="my-brand-class navbar-brand btn btn-link"
        disabled={false}
        onClick={[Function]}
        role="link"
        title="my brand title"
        type="button">
        Example App Name
      </button>
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button">
        <span
          className="sr-only">
          Toggle navigation
        </span>
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse">
      <ul
        className="nav navbar-nav"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="hello"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-bars"
              title={null} />
          </a>
        </li>
      </ul>
      <ul
        className="nav navbar-nav navbar-right"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="world"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-heart"
              title={null} />
          </a>
        </li>
        <li
          className="dropdown"
          style={undefined}>
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle"
            href=""
            id="myDropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button">
            user 1
             
            <span
              className="caret" />
          </a>
          <ul
            aria-labelledby="myDropdown"
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                href=""
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                <i
                  aria-hidden="true"
                  className="fa fa-fw fa-cog"
                  title={null} />
                settings
              </a>
            </li>
          </ul>
        </li>
      </ul>
      <div
        className="navbar-form navbar-right"
        onSubmit={[Function]}>
        <div
          className="form-group">
          <input
            className="form-control"
            id="globalSearch"
            onChange={[Function]}
            placeholder="search anything"
            type="text" />
        </div>
        <button
          className="btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-search"
            title={null} />
          Save
        </button>
      </div>
    </div>
  </div>
</nav>
`;

exports[`AppHeaderBar should render wihtout brandLink 1`] = `
<nav
  className="tc-app-header-bar navbar navbar-inverse navbar-fixed-top">
  <div
    className="container-fluid">
    <div
      className="navbar-header">
      <span
        className="navbar-brand">
        Example App Name
      </span>
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button">
        <span
          className="sr-only">
          Toggle navigation
        </span>
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse">
      <ul
        className="nav navbar-nav"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="hello"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-bars"
              title={null} />
          </a>
        </li>
      </ul>
      <ul
        className="nav navbar-nav navbar-right"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="world"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-heart"
              title={null} />
          </a>
        </li>
        <li
          className="dropdown"
          style={undefined}>
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle"
            href=""
            id="myDropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button">
            user 1
             
            <span
              className="caret" />
          </a>
          <ul
            aria-labelledby="myDropdown"
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                href=""
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                <i
                  aria-hidden="true"
                  className="fa fa-fw fa-cog"
                  title={null} />
                settings
              </a>
            </li>
          </ul>
        </li>
      </ul>
      <div
        className="navbar-form navbar-right"
        onSubmit={[Function]}>
        <div
          className="form-group">
          <input
            className="form-control"
            id="globalSearch"
            onChange={[Function]}
            placeholder="search anything"
            type="text" />
        </div>
        <button
          className="btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-search"
            title={null} />
          Save
        </button>
      </div>
    </div>
  </div>
</nav>
`;
